### PR TITLE
server: refactor kv.

### PR DESCRIPTION
**Pull request by disksing**
Extract kv to an interface. The benefits include a) plugin schedulers or namespace classifiers can save data without depend on server package, b) some tests can use in-memory storage without starting etcd server.

### CODE REVIEW

File: server/core/kv_base.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	Save(key, value string) error
+}
+
+type memoryKV struct {
+	sync.RWMutex
+	tree *btree.BTree
+}
+
+// NewMemoryKV returns an in-memory kvBase for testing.
+func NewMemoryKV() KVBase {
```
Comment (siddontang) [inline server/core/kv_base.go]: any test to test the KV too?

Comment (disksing) [reply server/core/kv_base.go]: Next PR will use it.

**Comment by nolouch** [comment]
LGTM

**Pull request merged**